plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -893,21 +893,27 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
                 verify(previous != newSnapshotId, "Failed to get new snapshot ID "));
 
         if (!computedStatistics.isEmpty()) {
-            icebergTable = catalog.loadTable(session, table.getName());
-            beginTransaction(icebergTable);
-            icebergTable = transaction.table();
-            CollectedStatistics collectedStatistics = processComputedTableStatistics(icebergTable, computedStatistics);
-            StatisticsFile statisticsFile = tableStatisticsWriter.writeStatisticsFile(
-                    session,
-                    icebergTable,
-                    newSnapshotId,
-                    INCREMENTAL_UPDATE,
-                    collectedStatistics);
-            transaction.updateStatistics()
-                    .setStatistics(newSnapshotId, statisticsFile)
-                    .commit();
-
-            transaction.commitTransaction();
+            try {
+                beginTransaction(catalog.loadTable(session, table.getName()));
+                Table reloadedTable = transaction.table();
+                CollectedStatistics collectedStatistics = processComputedTableStatistics(reloadedTable, computedStatistics);
+                StatisticsFile statisticsFile = tableStatisticsWriter.writeStatisticsFile(
+                        session,
+                        reloadedTable,
+                        newSnapshotId,
+                        INCREMENTAL_UPDATE,
+                        collectedStatistics);
+                transaction.updateStatistics()
+                        .setStatistics(newSnapshotId, statisticsFile)
+                        .commit();
+
+                transaction.commitTransaction();
+            }
+            catch (Exception e) {
```

Comment:
FYI, new logic. Now the stats writing is in try-catch

Reply:
@alexjo2144 @findinpath @losipiuk PTAL

Reply:
Would it be reasonable to have error counter exposed as JMX?
Do we we expose JMX from connectors btw?

Reply:
Yes, connectors can expose JMX stats. I don't know who would use it though.
Logs are IMO even easier to monitor.

@losipiuk should i log on ERROR or WARN?

Reply:
`Failed to write checkpoint for table` goes at ERROR level, so switching to ERROR here too

Reply:
So the pros/cons here is that we're stuck committing the new statistics in a separate transaction than the new data files, this could cause a race condition with another writer, but on the other hand if stats collection fails we still insert the data?

i'm not sure if it's worth it. If stats collection is causing writes to fail users can disable it using the config toggle.